be/src/olap/rowset/segment_v2/segment.cpp

```diff
@@ -148,7 +148,10 @@ Status Segment::_parse_footer() {
 
     uint8_t fixed_buf[12];
     size_t bytes_read = 0;
-    RETURN_IF_ERROR(_file_reader->read_at(file_size - 12, Slice(fixed_buf, 12), &bytes_read));
+    // whole / sub file cache will use it
+    io::IOContext io_ctx;
+    RETURN_IF_ERROR(
+            _file_reader->read_at(file_size - 12, Slice(fixed_buf, 12), &bytes_read, &io_ctx));
```

Comment:
check io_ctx != nullptr in read_at() is better.

Reply:
fixed